GeneratorInterface/Hydjet2Interface/src/Hydjet2Hadronizer.cc

```diff
@@ -40,32 +40,40 @@ using namespace edm;
 using namespace std;
 using namespace gen;
 
-int Hydjet2Hadronizer::convertStatusForComponents(int sta, int typ) {
-  if (sta == 1 && typ == 0)
-    return 6;
-  if (sta == 1 && typ == 1)
-    return 7;
-  if (sta == 2 && typ == 0)
-    return 16;
-  if (sta == 2 && typ == 1)
-    return 17;
-
+int Hydjet2Hadronizer::convertStatusForComponents(int sta, int typ, int pySta) {
+  int st = -1;
+  if (typ == 0)  //soft
+    st = 2 - sta;
+  else if (typ == 1)
+    st = convertStatus(pySta);
+
+  if (separateHydjetComponents_) {
+    if (st == 1 && typ == 0)
+      return 6;
+    if (st == 1 && typ == 1)
+      return 7;
+    if (st == 2 && typ == 0)
```

Comment:
This is different than what you had before this PR: is this the actual bug fix?
(The PR description is not clear about the origin of the bug: is there a github issue mentioning it? If so, better to put a link to it in the PR description)

Reply:
No, it was locally funded. The origin is related to different codes for decayed particles for type==0 and type==1. Within type==0 it was 0 for decayed. But we need to set it 2 for Geant (L46)